### PR TITLE
[14.0][MIG] partner_statement: Migration to 14.0

### PR DESCRIPTION
**Pull request by RujiaL**

### CODE REVIEW

File: partner_statement/report/report_statement_common.py [hunk 1]
```diff
@@ -44,15 +44,17 @@ def _show_buckets_sql_q1(self, partners, date_end, account_type):
                 ELSE l.balance + sum(coalesce(pc.amount, 0.0))
             END AS open_due,
             CASE WHEN l.balance > 0.0
-                THEN l.amount_currency - sum(coalesce(pd.amount_currency, 0.0))
-                ELSE l.amount_currency + sum(coalesce(pc.amount_currency, 0.0))
+                THEN l.amount_currency - sum(coalesce(pd.debit_amount_currency, 0.0))
```
Comment (gdgellatly) [inline partner_statement/report/report_statement_common.py:47]: @MiquelRForgeFlow Your advice here. The apr record has changed, but I don't have any v14.0 forex records to check what the right thing to do is, but my gut tells me this is wrong.

File: partner_statement/report/outstanding_statement.py [hunk 2]
```diff
@@ -35,14 +35,16 @@ def _display_lines_sql_q1(self, partners, date_end, account_type):
                 ELSE l.balance + sum(coalesce(pc.amount, 0.0))
             END AS open_amount,
             CASE WHEN l.balance > 0.0
-                THEN l.amount_currency - sum(coalesce(pd.amount_currency, 0.0))
-                ELSE l.amount_currency + sum(coalesce(pc.amount_currency, 0.0))
+                THEN l.amount_currency - sum(coalesce(pd.debit_amount_currency, 0.0))
```
Comment (gdgellatly) [inline partner_statement/report/outstanding_statement.py:38]: @MiquelRForgeFlow as per other comment

File: partner_statement/report/report_statement_common.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # Copyright 2018 ForgeFlow, S.L. (https://www.forgeflow.com)
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
 
+# flake8: noqa: C901: cannot find a simple way to do with the complex of "_get_report_values"
```
Comment (MiquelRForgeFlow) [inline partner_statement/report/report_statement_common.py]: put that just in the _get_report_values method instead

File: partner_statement/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Copyright 2018 ForgeFlow, S.L. (http://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Partner Statement",
+    "version": "14.0.1.1.0",
```
Comment (MiquelRForgeFlow) [inline partner_statement/__manifest__.py]:
```suggestion
    "version": "14.0.1.0.0",
```

File: partner_statement/report/report_statement_common.py (hunk 1 above)
Comment (MiquelRForgeFlow) [reply partner_statement/report/report_statement_common.py:47]: Ok, I will check.

**Comment by MiquelRForgeFlow** [comment]
please cherry pick https://github.com/OCA/account-financial-reporting/pull/778

File: partner_statement/report/report_statement_common.py (hunk 1 above)
Comment (gdgellatly) [reply partner_statement/report/report_statement_common.py:47]: @MiquelRForgeFlow when you have a chance :) or anyone else interested.

File: partner_statement/wizard/statement_common.py [hunk 5]
```diff
@@ -14,7 +14,7 @@ class StatementCommon(models.AbstractModel):
     def _get_company(self):
         return (
             self.env["res.company"].browse(self.env.context.get("force_company"))
```
Comment (gdgellatly) [inline partner_statement/wizard/statement_common.py]: @RujiaL I don't think this is correct in 14.0 anymore as force_company is deprecated. Probably just self.env.company as a default is fine and this function no longer needed.

**Review by jcdrubay — APPROVED** [review]
Tested and it works very nicely.

**Review by hkapatel-initos — CHANGES_REQUESTED** [review]
Hello @RujiaL Can you put all the changes together and make one commit for Migration to 14.0

**Comment by MiquelRForgeFlow** [comment]
please add https://github.com/OCA/account-financial-reporting/pull/804

**Comment by hkapatel-initos** [comment]
Hello @RujiaL Can you put all the changes together and make one commit for Migration to 14.0

**Comment by victoralmau** [comment]
Superseed by: https://github.com/OCA/account-financial-reporting/pull/826

**Pull request closed without merging**